import.gd

```diff
@@ -37,69 +71,70 @@ func createTileset(var data, var cell_size):
 			tiles = t["tiles"]
 		for y in range(0, height, cell_size.y):
 			for x in range(0, width, cell_size.x):
-				var xy = Vector2(x, y)
-				var rect = Rect2(xy, size)
-				ts.create_tile(count)
-				ts.tile_set_texture(count, texture)
-				ts.tile_set_region(count, rect)
-				var id = str(count - int(t["firstgid"]))
-
-				if t.has("tiles"):
-					for tile in tiles:
-						if tile == id and tiles[tile].has("objectgroup"):
-							for obj in tiles[tile]["objectgroup"]["objects"]:
-								if !obj.has("polyline") and !obj.has("polygon") and !obj.has("ellipse"):
-									var w = obj["width"]
-									var h = obj["height"]
-									var xx = obj["x"]
-									var yy = obj["y"]
-									var rectshape = RectangleShape2D.new()
-									rectshape.set_extents(Vector2(w/2, h/2))
-									ts.tile_set_shape(count, rectshape)
-									ts.tile_set_shape_offset(count, Vector2(w/2 + xx, h/2 + yy))
-								elif obj.has("ellipse"):
-									var w = obj["width"]
-									var h = obj["height"]
-									var xx = obj["x"]
-									var yy = obj["y"]
-									if w == h:
-										var circleshape = CircleShape2D.new()
-										circleshape.set_radius(w/2)
-										ts.tile_set_shape(count, circleshape)
+				var intcount = int(count)
+				if ((onlyused == false) || ((onlyused == true) && ( corres_array.find(intcount) != -1))): #when onlyused is true, the tile from Tiled tileset is imported in Godot tileset only if it is used in the map
```

Comment:
Instead of `onlyused == false` use just `!onlyused`. And just `onlyused` for `onlyused == true`. It works best this way and otherwise it does not import any tile unless `onlyused` is strictly `true` or `false`. And if the user didn't touch the checkbox, it's value is `nil` which is neither `true` nor `false`, so no tile is imported.